sw/device/lib/testing/json/i2c_target.h

```diff
@@ -24,6 +24,7 @@ UJSON_SERDE_STRUCT(I2cTargetAddress, i2c_target_address_t, STRUCT_I2C_TARGET_ADD
     field(length, uint8_t) \
     field(address, uint8_t) \
     field(continuation, uint8_t) \
+    field(repeated_start, bool) \
     field(data, uint8_t, 256)
 UJSON_SERDE_STRUCT(I2cTransaction, i2c_transaction_t, STRUCT_I2C_TRANSACTION);
```

Comment:
This totally has the wrong name for an I2C struct, heh. It's clearly neither a "transfer" nor a "transaction".

The API would more closely map to I2C terms if we had these two commands:

```c
// Should be used to begin every I2C transfer.
// Issues a Start, address, command, and optional data and Stop conditions.
// A 0 data_len means there is no data.
#define STRUCT_I2C_TRANSFER_START(field, string) \
    field(data_len, uint8_t) \
    field(addr_cmd, uint8_t) \
    field(stop, bool) \
    field(data, uint8_t, 255)
UJSON_SERDE_STRUCT(I2cTransferStart, i2c_transfer_start_t, STRUCT_I2C_TRANSFER_START);

// For long transfers, used to add more data to the transfer begun with
// I2cTransferStart. A 0 data_len indicates 256 bytes.
#define STRUCT_I2C_TRANSFER_CONT(field, string) \
    field(data_len, uint8_t) \
    field(stop, bool) \
    field(data, uint8_t, 256)
UJSON_SERDE_STRUCT(I2cTransferCont, i2c_transfer_cont_t, STRUCT_I2C_TRANSFER_CONT);
```

Then, every transfer begins with I2cTransferStart, and the last transfer of the transaction ends with `stop = true` (instead of the next transfer's I2cTransferStart).

Reply:
I refactored the test to use the suggested Transfer struct, but I only implemented the start command for now as it is enough for the existing tests.

Reply:
You know, somehow I totally missed that this file is for target mode. With respect to the API suggestion, sorry I lost the context!

I guess it can work either way, though. For the target case, these would be reports of what was observed to have happened (a passive monitor instead of an active driver).

Reply:
Yes, it works either way.